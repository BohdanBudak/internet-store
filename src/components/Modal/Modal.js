import React from 'react';
import Button from "../Button/Button";
import injectSheet from "react-jss";
import styles from "./ModalStyles";
import PropTypes from 'prop-types';

const Modal = (props) => {

    const stopCloseModal = (e) => {
        e.stopPropagation()
    }

    const {header, closeButton, text, actions, classes, onClose} = props;

    return (
        <div className={classes.modal} onClick={onClose}>
            <div className={classes.modalContainer} onClick={stopCloseModal}>
                <div className={classes.modalHeader}>
                    <h2 className={classes.modalHeaderTitle}>
                        {header}
                    </h2>
                    {closeButton === "true" &&
                    <div className={classes.modalBtnClose}>
                        <Button text="X" width={'80'} height={'40'} fontSize={'18'} onClick={onClose}/>
                    </div>
                    }
                </div>
                <div className={classes.modalBody}>
                    <p className={classes.modalBodyText}>
                        {text}
                    </p>
                </div>
                {actions.length !== 0 &&
                <div className={classes.modalFooter}>
                    {actions}
                </div>
                }
            </div>
        </div>
    );
}

Modal.propTypes = {
    classes: PropTypes.object,
    header: PropTypes.string,
    closeButton: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.bool
    ]),
    text: PropTypes.string,
    actions: PropTypes.array,
    onClose: PropTypes.func,
};

export default injectSheet(styles)(Modal);