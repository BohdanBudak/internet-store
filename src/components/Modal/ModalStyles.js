export default {

    modal: {
        position: 'absolute',
        top: 0,
        left: 0,
        display: 'flex',
        width: '100%',
        height: '100%',
        color: 'white',
        backgroundColor: 'rgba(0, 0, 0, 0.7)',
    },

    modalContainer: {
        position: 'fixed',
        top: '45%',
        left: '35%',
        width: '450px',
        height: '200px',
        backgroundColor: 'darkslategray',
        borderRadius: '5px',
        zIndex: '1',
    },

    modalHeader: {
        position: 'relative',
        width: '100%',
        textAlign: 'center',
        backgroundColor: 'rgba(255, 255, 255, 0.3)',
    },

    modalHeaderTitle: {
        margin: 0,
        padding: '10px',
        fontSize: '18px',
    },

    modalBtnClose: {
        position: 'absolute',
        top: 0,
        right: 0,
        backgroundColor: 'rgba(255, 255, 255, 0.2)',
        borderRadius: 0,
        cursor: 'pointer',
    },

    modalBody: {
        padding: '0 50px',
    },

    modalBodyText: {
        margin: '30px 0',
        fontSize: '16px',
    },

    modalFooter: {
        display: 'flex',
        justifyContent: 'space-around',
        alignItems: 'center',
        padding: '20px 0',
    }
}