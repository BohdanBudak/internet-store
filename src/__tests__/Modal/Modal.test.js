import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { shallow, render } from 'enzyme';
import Modal from '../../components/modal/Modal'

describe('Initial Modal', () => {

    const props = {
        header: "",
        text: "",
        closeButton: false,
        actions: [],
        onClose: () => {}
    };

    describe('Modal text props', () => {

        it('should render a Modal with header', () => {
            const nextProps = {
                ...props,
                header: "Modal Header"
            };

            const modalComponent = render(<Modal {...nextProps}/>);
            expect(modalComponent.text()).toEqual('Modal Header');
        });

        it('should render a Modal with text', () => {
            const nextProps = {
                ...props,
                text: "Modal Main text"
            };

            const modalComponent = render(<Modal {...nextProps}/>);
            expect(modalComponent.text()).toEqual('Modal Main text');
        });
    });

    describe('Modal props Button close', () => {

        it('should render a Modal without close Button', () => {

            const modalComponent = shallow(<Modal {...props}/>);
            expect(modalComponent.find('div.modalHeader').children.length).toBe(1);
        });

    });
});