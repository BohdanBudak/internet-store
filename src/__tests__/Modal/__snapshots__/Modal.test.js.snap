// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial Modal Modal props function onClose renders properly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "Modal-modal-0-2-1",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "Modal-modalContainer-0-2-2",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "Modal-modalHeader-0-2-3",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "Modal-modalHeaderTitle-0-2-4",
                },
                "children": Array [],
                "name": "h2",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "Modal-modalBody-0-2-6",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "Modal-modalBodyText-0-2-7",
                  },
                  "children": Array [],
                  "name": "p",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "Modal-modalBody-0-2-6",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "Modal-modalBodyText-0-2-7",
                },
                "children": Array [],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "Modal-modalHeader-0-2-3",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "Modal-modalHeaderTitle-0-2-4",
                  },
                  "children": Array [],
                  "name": "h2",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
